src/makeCalculator.js

```diff
@@ -37,7 +37,45 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(number, thisLink) {
+      thisLink.result += number;
+    },
+
+    subtract(number, thisLink) {
+      thisLink.result -= number;
+    },
+
+    multiply(number, thisLink) {
+      thisLink.result *= number;
+    },
+
+    divide(number, thisLink) {
+      if (number === 0) {
+        throw new Error('Division by zero!');
+      }
+
+      thisLink.result /= number;
+    },
+
+    operate(operation, number) {
+      const thisLink = this;
+
+      operation(number, thisLink);
```

Comment:
```suggestion
      operation.call(this, number);
```

Reply:
Thank you! I have not made it to 'function.call' yet.